### DISCUSSION TITLE
[Feature Request] Panel time affected by Value Mapping

### DISCUSSION THREAD

**Discussion by imReker** [grafana/grafana#41130] [Dashboards, Panels & Visualizations]
**What happened**:
Get wrong time when unix timestamp matches Value Mapping.

**What you expected to happen**:
Time is not affected by Value Mapping.

**How to reproduce it (as minimally and precisely as possible)**:
1. Set a big Range Value Mapping, e.g., 1000-9635389520000 map to text 'HIGH'.
2. Time in the panel displayed as 'HIGH'.

**Environment**:
- Grafana version: Grafana v8.2.2-39068 (183fe28769)
- User OS & Browser: Windows 11 Edge

**Comment by zuchka** [comment]
Thanks for creating this issue, @imReker 

Yes, I see the behavior. Here is a dashboard where value mapping has overwritten a timestamp:

https://play.grafana.org/d/jDwXddKnz/value-mapping-a-range-overwrites-timestamps?orgId=1

![CleanShot 2021-10-29 at 17 11 55@2x](https://user-images.githubusercontent.com/37156449/139518318-06c9aa50-b7c3-42ee-8c82-15fa7f81b2da.png)

**Comment by hugohaggmark** [comment]
Thanks for reporting this @imReker and thanks for repro @zuchka. Would you both expect the time values to be excluded from all value mappings? I'm just trying to figure out what you expect from this issue?

**Comment by imReker** [comment]
> Would you both expect the time values to be excluded from all value mappings?

In my opinion, the default Value Mapping settings should exclude the Time, because:
1. It's 'value' mapping. Time(underlying unix timestamp),  intuitively is not a `value`.
2. It's rare to map a exact time stamp (value/range) to some other thing.

Can we default exclude Value Mapping to Time, and allow users manually add an Value Mapping override to Time values to keep the compatibility with the old behavior.

**Comment by hugohaggmark** [comment]
@imReker aha, you mean that this used to work before? Can you identify which Grafana version when this stopped working?

**Comment by imReker** [comment]
No, the 'old' I mean the current behavior, which is Value mapping applied to Time.

**Comment by hugohaggmark** [comment]
I see ok, I'm converting this issue to a discussion so we can better track it.

**Comment by torkelo** [comment]
think this looks like a bug, value mappings should by default apply to time fields?

**Reply by hugohaggmark** [reply]
Do you mean that time fields should always be excluded by Value Mappings @torkelo?

**Reply by torkelo** [reply]
not sure, but a possible option: https://github.com/grafana/grafana/pull/41132

**Reply by imReker** [reply]
@torkelo  Maybe instead of just ignoring the Timestamp, passing the formatted Timestamp string (by setting Unit) to value mapping is better?
Use case:
1. Colorize the Time to red every Friday / every 10:00 / every 1st day of month in Table by Regex value mapping.
2. Customize the weekday name, Monday -> MON/Mon/1 by String value mapping.

**Comment by KnpA** [comment]
Hello,
I got a use case for value mapping on Time field.
I receive a "default" date from an ElasticSearch datasource (value : 0001-01-01T00:00:00Z) and today I can't handle it in a nice way with a custom unit (time: DD/MM/YYYY HH:mm:ss) + value mapping (01/01/0001 00:00:00 => -) to hide this default value to end use.
My ugly workaround:
 
![image](https://user-images.githubusercontent.com/6367927/181276165-c3a20e38-99b1-4631-b462-da6a20c3447c.png)

Why ugly ? => I can't handle the timezone for the "real" values shown to the end user.
And I also have no other workaround, like "transform" option to pre-format my time field to a string that could then be value-mapped

**Comment by natellium** [comment]
Migrated this to an issue https://github.com/grafana/grafana/issues/78627
